packages/dds/sequence/api-report/sequence.api.md

```diff
@@ -370,6 +371,31 @@ export interface ISharedIntervalCollection<TInterval extends ISerializableInterv
     getIntervalCollection(label: string): IIntervalCollection<TInterval>;
 }
 
+// @alpha (undocumented)
+export interface ISharedSegmentSequence<T extends ISegment> extends ISharedObject<ISharedSegmentSequenceEvents>, ISharedIntervalCollection<SequenceInterval>, MergeTreeRevertibleDriver {
```

Comment:
I added all the members currently required by code in this repo. There are some more public methods that might belong here, and some that I had to add which should probably be removed.

Reply:
Most functions that exist here are reasonable.
- getCurrentSeq is probably not something we want to expose, but it's not the end of the world and I don't think we should attempt to fix that here
- groupOperation is as you said. would be nice to remove our usage so we can remove the api. but again, not here.
- initializeLocal is something we technically did expose but shouldn't be used by customers, and isn't a SharedSegmentSequence thing (so if we wanted to put it somewhere because we were concerned about breaking customers, the right place would be ISharedObject not here, but I think it's probably fine to just remove)

We definitely also want the following:

- `annotateRange`
- `removeRange`

seems like we might have a test gap in terms of using our public API (we definitely have tests that exercise those functions, but I'm guessing they end up typed to the concrete class so you didn't notice)




This leaves a few APIs:

- getRangeExtentsOfPosition: Seems like there might be some not very extensive usage of this today based on a search, but it's not really something we need to implement. it can be implemented with other methods on the public API. i'd be ok with either decision.
- insertFromSpec: don't think anyone uses this publicly. seems good to remove.
- resolveRemoteClientPosition: This looks like it's used today to support some presence scenarios using signals. we probably want it.
- posFromRelativePos: If we're including the relative pos methods on SharedString, it seems like we should probably have this as well. That said, i'd love if we could remove all of them as I don't think support is good and not sure anyone's using.

Reply:
tl;dr is that I think you probably can't get away with removing most of the previously public things on the `SharedSegmentSequence` class, and I think it might be a bad idea to try to do so as part of a change aimed at hiding the class (it's probably better to do that sort of thing in follow-ups with more targeted guidance on API surface reduction)

Reply:
@anthony-murphy might have some thoughts here.

Thanks for doing this though in any case. glad to see progress here.

Reply:
A lot of tests fail to build without initializeLocal here, so if we want to remove it, thats significant work probably best done separately.

There is also `getIntervalCollection`. As its mentioned in the docs for getIntervalCollectionLabels, I've added that along with the ones you mentioned. Thats all the public members from SharedSegmentSequence. I put some in a `/ #region APIs we might want to remove` to help track them for later.